lib/cocoapods/installer/xcode/pods_project_generator.rb

```diff
@@ -294,10 +329,15 @@ def wire_target_dependencies(target_installation_results)
                   end
                   test_native_target.add_dependency(dependency_installation_result.native_target)
                   add_framework_file_reference_to_native_target(test_native_target, pod_target, test_dependent_target, frameworks_group)
-                  test_spec_consumers = test_specs.map { |test_spec| test_spec.consumer(pod_target.platform) }
-                  if test_spec_consumers.any?(&:requires_app_host?)
-                    app_host_target = project.targets.find { |t| t.name == pod_target.app_host_label(test_specs.first.test_type) }
-                    test_native_target.add_dependency(app_host_target)
+                  # Wire app host dependencies to test native target
+                  if pod_target.test_spec_consumers.any?(&:requires_app_host?)
+                    pod_target.supported_test_types.each do |test_type|
+                      app_host_target = app_hosts_by_host_key.select do |app_host_key|
```

Comment:
I think you can use `find` here instead of `select`

Reply:
they behave slightly differently but `find` won't actually do what we need. it's weird to explain so here's a sample:

```ruby
a = {:foo => "bar", :a => "b"}

a.find {|a| true}
# [:foo, "bar"]

a.select {|a| true}
# {:foo => "bar", :a => "b}
```

Reply:
ah I was wrong

Reply:
it's all good! this behavior is extremely unintuitive and not even documented! 🙃

Reply:
didn't know that two structs with identical values were equivalent in ruby. learned something new today